config/ompi_setup_mpi_fortran.m4

```diff
@@ -54,6 +55,26 @@ AC_DEFUN([OMPI_SETUP_MPI_FORTRAN],[
     OMPI_FORTRAN_HAVE_F08_ASSUMED_RANK=0
     OMPI_FORTRAN_HAVE_PRIVATE=0
 
+    AC_MSG_CHECKING([if we should mark mpif.h bindings as deprecated])
+    AC_ARG_ENABLE([deprecate-mpif-h],
+                  [AS_HELP_STRING([--enable-deprecate-mpif-h],
+                                  [Mark the mpif.h bindings as deprecated (default: enabled)])])
+    if test "x$FC" != "xflang-new"; then
+        AC_MSG_RESULT([no (only for flang-new)])
+        OMPI_FORTRAN_DEPRECATE_MPIF_H=""
+    elif test "$enable_deprecate_mpif_h" = "no"; then
+        AC_MSG_RESULT([no])
+        OMPI_FORTRAN_DEPRECATE_MPIF_H=""
+    else
+        if test "$enable_deprecate_mpif_h" = "yes"; then
+            AC_MSG_RESULT([yes])
+        else
+            AC_MSG_RESULT([yes (default)])
+        fi
+        OMPI_FORTRAN_DEPRECATE_MPIF_H="#warning mpif.h is deprecated since MPI 4.1. Refer to MPI Sec. 19.1.4."
+    fi
+    AC_SUBST(OMPI_FORTRAN_DEPRECATE_MPIF_H)
```

Comment:
Shouldn't you just test if the directive works with the current Fortran compiler?

Reply:
it is an interesting test, but i suppose if the output contains `mpif.h is deprecated since MPI 4.1. Refer to MPI Sec. 19.1.4.` then that is a reasonable definition of "works"